cpp/src/arrow/compute/kernels/scalar_validity.cc

```diff
@@ -307,6 +379,11 @@ const FunctionDoc is_nan_doc("Return true if NaN",
                              ("For each input value, emit true iff the value is NaN."),
                              {"values"});
 
+const FunctionDoc indices_nonzero_doc{
+    "Compare to zero or false",
```

Comment:
The summary line is a bit vague, how about just using the sentence below?

Reply:
:+1: